code/workspaces/infrastructure-api/src/kubernetes/serviceAccountApi.js

```diff
@@ -1,11 +1,29 @@
 import logger from '../config/logger';
 import { getCoreV1Api } from './kubeConfig';
 
+async function namespacedServiceAccountExists(name, namespace) {
+  const k8sApi = getCoreV1Api();
+  logger.info(`Checking serviceAccount ${name} exists in ${namespace}`);
+  try {
+    await k8sApi.readNamespacedServiceAccount(name, namespace);
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+/* eslint-disable consistent-return */
 async function createNamespacedServiceAccount(name, namespace) {
+  if (await namespacedServiceAccountExists(name, namespace)) {
+    logger.info(`ServiceAccount ${name} already exists in namespace ${namespace}`);
+    return true;
```

Comment:
It feels a bit odd to `return true;` here if you don't care about the return value elsewhere (nothing else is being returned). I would be tempted to just `return;` but I don't know whether this is bad practice or something...

Reply:
Yes that makes sense, doesn't really matter what gets returned.